analysis/quantifier-comparisons/02-compare_quants.Rmd

```diff
@@ -0,0 +1,350 @@
+
+```{r }
+hub = AnnotationHub::AnnotationHub(ask = FALSE)
+# Ensembl v100 Homo sapiens is AH79689
+ensdb = hub[["AH79689"]]
+ensg <- genes(ensdb)
+```
+
+Create vectors of mitochondrial genes and coding genes for later.
+
+```{r}
+# create the mitochondrial gene list
+mito_genes <- ensg[seqnames(ensg) == 'MT']$gene_id
+
+coding_genes <- ensg[ensg$gene_biotype == "protein_coding"]$gene_id
+```
+
+## Calculate cell QC statistics
+
+A small function to set default QC calculations for SCEs in a list:
+```{r}
+addPerCellQC_mito <- function(sce, mito = mito_genes){
+  scater::addPerCellQC(
+    sce, 
+    subsets = list(mito = mito[mito %in% rownames(sce)])
+  )
+}
+```
+
+Calculate cell QC statistics for each set of SCES
+
+```{r}
+alevin_sces <- alevin_sces %>% 
+  purrr::map(addPerCellQC_mito)
+
+kallisto_sces <- kallisto_sces %>% 
+  purrr::map(addPerCellQC_mito)
+
+cellranger_sces <- cellranger_sces %>% 
+  purrr::map(addPerCellQC_mito)
+```
+
+Make a data frames of the cell QCs.
+```{r}
+celldata_to_df <- function(sce){
+  as.data.frame(colData(sce)) %>%
+  tibble::rownames_to_column(var = "cell_id")
+}
+
+alevin_cell_qc <- alevin_sces %>%
+  purrr::map_df(celldata_to_df, .id = "quant_id")
+
+kallisto_cell_qc <- kallisto_sces %>%
+  purrr::map_df(celldata_to_df, .id = "quant_id")
+
+cellranger_cell_qc <- cellranger_sces %>%
+  purrr::map_df(celldata_to_df, .id = "quant_id")
+
+# combine all the data frames into one
+cell_qc <- dplyr::bind_rows(
+  alevin = alevin_cell_qc,
+  kallisto = kallisto_cell_qc,
+  cellranger = cellranger_cell_qc,
+  .id = "tool"
+) %>%
+  dplyr::left_join(quant_info,
+                   by = c("tool" = "tool", 
+                          "quant_id" = "quant_dir"))
+```
+
+## Comparing summary stats across tools
+
+Plot distributions of Total UMI counts
+```{r}
+ggplot(cell_qc, aes(x = sum, color = sample)) +
+  geom_density() +
+  xlab("Total UMI count") +
+  facet_grid(tool ~ decoy)
+```
+
+Plot distributions of Detected gene counts
+```{r}
+ggplot(cell_qc, aes(x = detected, color = sample)) +
+  geom_density() +
+  xlab("Detected genes") +
+  facet_grid(tool ~ decoy)
+```
+Kallisto tends to detect more UMIs and more transcripts per cell. 
+
+Lets make a little table:
+```{r}
+cell_qc %>%
+  dplyr::group_by(sample, tool, decoy) %>%
+  dplyr::summarize(dplyr::across(
+    c(sum, detected, subsets_mito_percent),
+    mean)) %>%
+  dplyr::rename("mean UMI count" = "sum", 
+                "mean detected genes" = "detected", 
+                "mean mitochondiral percent" = "subsets_mito_percent")
+```
+
+
+So, the naive conclusion is that Kallisto is mapping many more UMIs per cell, and capturing more genes (which reduces the mitochondrial percentage).
+
+Is this because we are excluding low count cells in the knee plot? 
+```{r}
+cell_qc %>%
+  dplyr::group_by(sample, tool, decoy) %>%
+  dplyr::tally()
+```
+Ugh, maybe. Cellranger consistently keeps the most cells, with kallisto the fewest.
```

Comment:
I can't make a suggestion here because of the code blocks, but _if I could_ I would add blank lines around this block.